Rewrite security section (#322)
akeneo/pim-api-docs: content/rest-api/permissions.md

# Permissions

In some connector use cases, you will need to restrict the access to the API.
There are two ways to handle this in the API, depending on the wanted result:
- if you want to restrict the access to some API endpoints, use [the ACLs authorizations](#endpoint-access-restrictions).
- if you want to restrict the access to certain parts of your product catalog, use the [Entreprise Edition permissions](#catalog-permissions-ee-only) _(only available in the Entreprise Edition)_.

> We detail in the sections below those two ways.

Author: ```suggestion
Please see below for more details about both permission systems.
```

> Please see below for more details about both systems. 

Author: ```suggestion
Please see below for more details about both permission systems. 
```

## Endpoint access restrictions

Like when using the PIM through the UI, we use ACLs to define which endpoint a connection can or cannot use.

Here are the simple steps to follow to configure those permissions.

1. Log into your favorite PIM.
1. Navigate to the `System/Roles` menu.
1. Click on the `Create role` button.
1. Input a new name for the user role you are creating, `ERP connection user role` for example.
1. In the `Users` tab, select the [API user you created earlier](/documentation/authentication.html#api-user-creation).
1. In the `Web API permissions` tab, select the permissions you want to give to your API user.

::: versions id="web_api_permission" 2.x![Web API permission tab screenshot](/img/rest-api/activate_api_access.png) 1.7![Web API permission tab screenshot](/img/rest-api/v1.7/activate_api_access.png)
:::

We strongly recommend you to create **dedicated user roles** for your API users, different from the user roles that you use for your UI users. So go ahead and create one user role for every API connection you will need.

::: warning
Do not give any UI permissions to your API user roles.
:::

### Overall access

> The first ACL `Overall Web API access` means that if enabled, each user depending on that role will have access to the Web API.

Author: ```suggestion
The first ACL `Overall Web API access` means that if enabled, each user under that role will have access to the Web API.
```

1. Input a new name for the user group you are creating, `ERP connection user group` for example.
![New ERP connection user group](/img/rest-api/erp-connection-user-group.png)
1. In the `Users` tab, select the [API user you created earlier](/documentation/authentication.html#api-user-creation).
![API user in the user group](/img/rest-api/my-erp-user-in-group.png)
1. Then, navigate to the `Settings/Categories` menu.
1. For each category you wish to only create proposals, go in the `Permissions` tab.
1.  Add the group you just created into the `Allowed to view products` and `Allowed to edit products` inputs.
1. If your user group is already set into the `Allowed to own products` input, remove it. 
![Permissions for proposals](/img/rest-api/proposal-permission-mode.png)
1. Don't forget to click on the `Save` button.

That's it! :tada:

Don't hesitate to test your configuration by calling the [PATCH product endpoint](/api-reference.html#patch_products__code_) with an update in the body of your product. A draft of your product will be created.  
You will then need to send your draft for approval by using [this API endpoint](/api-reference.html#post_proposal).  

> If it works correctly, the owners of the categories in which is classified the product you try to update earlier, will receive in their Proposals screen, a new proposal like the one below.

Reviewer: ```suggestion
If it works correctly, the owners of the categories where the product you tried to update earlier is classified will receive a new proposal like the one below in their Proposals screen.
```

![Proposal generated by the API](/img/rest-api/proposal-by-api.png)

::: info
Proposals can be created only on product updates, not on product creations. :wink:
:::

::: tips
Having trouble creating proposals on one given product?   
As soon as your product is at least owned in one category by your API user group, the API will directly apply the updates, instead of creating a draft and then a proposal.  

> So be sure to verify that your product is only classified in categories for which your API user group does not own it.

Reviewer: ```suggestion
So make sure your product is only classified in categories that are not owned by your API user group. 
```

Reviewer: I'm really not sure about the content...
